### PR TITLE
fix: remove collector operator & add custom collector auth

### PR DESCRIPTION
**Pull request by ilinzweilin**

### CODE REVIEW

File: src/core/borrower/collect/test/collector.t.sol [hunk 1]
```diff
@@ -82,6 +83,7 @@ contract CollectorTest is DSTest {
     }
 
     function testSeizeCollectAnyUser() public {
+        collector.relyCollector(address(this));
```
Comment (manuelpolzhofer) [inline src/core/borrower/collect/test/collector.t.sol]: This is an edge case.

The test is `ward` and `investor` add the same time.

maybe a `UserCollector` with a `doCollect`?
